taskvine/src/worker/vine_transfer_server.c

```diff
@@ -51,11 +53,38 @@ static void vine_transfer_handler( struct link *lnk, struct vine_cache *cache )
 
 static void vine_transfer_process( struct vine_cache *cache )
 {
+	static int child_count = 0;
+	
+	/* 
+	If link is real, fork. Check if we are at the max
+	child count. If we are over, or link_accept timed out,
+	take the time to wait() on all exited children
+	*/
 	while(1) {
-		struct link *lnk = link_accept(transfer_link,time(0)+60);
-		if(lnk) {
-			vine_transfer_handler(lnk,cache);
-			link_close(lnk);
+		struct link *lnk = link_accept(transfer_link,time(0)+10);
+
+		child_count++;
+		pid_t p = fork();
+		if(p==0)
+		{	
+			if(lnk)
+			{
+				vine_transfer_handler(lnk,cache);
+				link_close(lnk);
+			}
+			_exit(0);
+		} 
+		else 
+		{
+			if(child_count < VINE_TRANSFER_PROC_MAX_CHILD) {
```

Comment:
Hmm, this isn't quite right, you are skipping the `waitpid` and going back to the `accept`, which is going to result in an odd accumulation of zombies.

Try it this way:
- `count< limit` Do a non-blocking `waitpid` and go back to `accept`
- `count>=limit` ?  Do a blocking waitpid until a child finishes.